Lib/dataclasses.py

```diff
@@ -1305,12 +1309,8 @@ def _add_slots(cls, is_frozen, weakref_slot, defined_fields):
     if qualname is not None:
         newcls.__qualname__ = qualname
 
-    if is_frozen:
-        # Need this for pickling frozen classes with slots.
-        if '__getstate__' not in cls_dict:
-            newcls.__getstate__ = _dataclass_getstate
-        if '__setstate__' not in cls_dict:
-            newcls.__setstate__ = _dataclass_setstate
+    if '__reduce__' not in cls_dict:
```

Comment:
no, i don't think so, because we need the new reduce function no matter if the dataclass is frozen or not.

however, looking at the tests, it looks like maybe `__getstate__` and `__setstate__` are required if the dataclass has slots?

Reply:
Pondering... I wonder if always having a `__reduce__` has meaningful performance implications?